lambdas/evaluate-gpg45-scores/src/main/java/uk/gov/di/ipv/core/evaluategpg45scores/gpg45/Gpg45ProfileEvaluator.java

```diff
@@ -0,0 +1,147 @@
+package uk.gov.di.ipv.core.evaluategpg45scores.gpg45;
+
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
+import com.nimbusds.jose.shaded.json.JSONArray;
+import com.nimbusds.jose.shaded.json.JSONObject;
+import com.nimbusds.jwt.SignedJWT;
+import uk.gov.di.ipv.core.evaluategpg45scores.domain.CredentialEvidenceItem;
+import uk.gov.di.ipv.core.evaluategpg45scores.domain.CredentialEvidenceItem.EvidenceType;
+import uk.gov.di.ipv.core.evaluategpg45scores.exception.UnknownEvidenceTypeException;
+import uk.gov.di.ipv.core.evaluategpg45scores.validation.FraudEvidenceValidator;
+import uk.gov.di.ipv.core.evaluategpg45scores.validation.KbvEvidenceValidator;
+import uk.gov.di.ipv.core.evaluategpg45scores.validation.PassportEvidenceValidator;
+import uk.gov.di.ipv.core.library.helpers.LogHelper;
+
+import java.text.ParseException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static uk.gov.di.ipv.core.library.domain.VerifiableCredentialConstants.VC_CLAIM;
+import static uk.gov.di.ipv.core.library.domain.VerifiableCredentialConstants.VC_EVIDENCE;
+
+public class Gpg45ProfileEvaluator {
+
+    private static final Gson gson = new Gson();
+    private static final int NO_SCORE = 0;
+
+    public boolean anyCredentialsGatheredDoNotMeetM1A(List<String> credentials)
```

Comment:
Just trying to understand, what's the difference between this and `!credentialsSatisfyProfile(credentials, Gpg45Profile.M1A)`? One is checking the credentials overall meet the profile whereas the other is checking that each credential alone meets the profile?

Reply:
This is a shortcut that is relevant right now, but will be taken out later. This is what Chris was talking about , starting "We need to fail a journey if we receive a VC from a CRI that won't allow us to meet M1A. ..."

Reply:
So this is a temporary thing to replace the validatecricheck lambda. If `anyCredentialsGatheredDoNotMeetM1A`  returns true in the handler we fail the journey immediately. So it's effectively checking if the last credential gathered was below score or had CI's and halting the journey at that point like we do at the moment. We can remove the validatecricheck lambda once this is being used.

The `credentialsSatisfyProfile` method will return true if we have enough pieces of evidence to meet the desired profile. It doesn't care if there is evidence that has been gathered that _doesn't _ meet the scores, or has CIs, because this lambda isn't responsible for knowing if there are other CRIs available to go get more evidence from. That's the job of the CRI selector lambda.

So in future, the `anyCredentialsGatheredDoNotMeetM1A` check will get effectively get done by the CRI selector. If we haven't met the gpg45profile yet, this new lambda will pass on to the cri selector lambda. If that lambda decides there's no other paths for a user to take (other CRIs) to try to gather evidence to meet the profile, the user will fail. It allows for more complex journeys in future.